lib/src/spec/validationEngine.spec.ts

```diff
@@ -6,168 +6,198 @@ import { FieldValidationResult } from '../entities';
-
-        //Act
-        let result = validationEngine.isFieldKeyMappingDefined(key);
-
-        //Assert
-        expect(result).to.be.false;
-    });
-
-    it('should returns isFieldKeyMappingDefined equals true ' +
-        'when passing a key equals "test" and validationsPerField equals ["test": validationFn]', () => {
-        //Arrange
-        let key = "test";
-        let validationFn = (vm, value) => {
-            let fieldValidationResult = new FieldValidationResult();
-            return Promise.resolve(fieldValidationResult);
-        }
-        let validationEngine = new ValidationEngine();
-        validationEngine.addFieldValidationAsync(key, validationFn);
-
-        //Act
-        let result = validationEngine.isFieldKeyMappingDefined(key);
-
-        //Assert
-        expect(result).to.be.true;
-    });
+        done();
+      });
+
+    // Assert
+    expect(formValidationBase.isValidationInProgress()).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals false ' +
+      'when passing a key equals undefined and validationsPerField equals []', () => {
+    //Arrange
+    let key = undefined;
+    let validationEngine = new ValidationEngine();
+
+    //Act
+    let result = validationEngine.isFieldKeyMappingDefined(key);
+
+    //Assert
+    expect(result).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals false ' +
+      'when passing a key equals null and validationsPerField equals []', () => {
+    //Arrange
+    let key = null;
+    let validationEngine = new ValidationEngine();
+
+    //Act
+    let result = validationEngine.isFieldKeyMappingDefined(key);
+
+    //Assert
+    expect(result).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals false ' +
+      'when passing a key equals "test" and validationsPerField equals []', () => {
+    //Arrange
+    let key = "test";
+    let validationEngine = new ValidationEngine();
+
+    //Act
+    let result = validationEngine.isFieldKeyMappingDefined(key);
+
+    //Assert
+    expect(result).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals false ' +
+      'when passing a key equals "test1" and validationsPerField equals ["test2": validationFn]', () => {
+    //Arrange
+    let key = "test1";
+    let validationFn = (vm, value) => {
+        let fieldValidationResult = new FieldValidationResult();
+        return Promise.resolve(fieldValidationResult);
+    };
+    let validationEngine = new ValidationEngine();
+    validationEngine.addFieldValidationAsync("test2", validationFn);
+
+    //Act
+    let result = validationEngine.isFieldKeyMappingDefined(key);
+
+    //Assert
+    expect(result).to.be.false;
+  });
+
+  it('should returns isFieldKeyMappingDefined equals true ' +
+      'when passing a key equals "test" and validationsPerField equals ["test": validationFn]', () => {
+    //Arrange
+    let key = "test";
```

Comment:
Ok, then is not on scope. That has to be done on separate issue. On my point of view, you can go ahead with merge.